### PR TITLE
feat(group-label): Add Label support

### PR DESCRIPTION
**Pull request by JTBlanchard**

### CODE REVIEW

File: thousandeyes/schemas.go
Comment (william20111) [inline thousandeyes/schemas.go:23]: single `-` here?

File: thousandeyes/schemas.go
Comment (william20111) [inline thousandeyes/schemas.go:428]: same as above?

File: thousandeyes/schemas.go
Comment (JTBlanchard) [reply thousandeyes/schemas.go:23]: I'm open to doing this differently, but there is some intention behind this.  This is a -- to indicate that it is not a part of the field name from the API's perspective.

Fields that have the same name are generally treated the same across tests.  But we have a few tests for which this is not the case.  My first thought was to just change these within the resource files for tests that are the exceptions.  But there are a couple of cases where more than one test uses the alternate behavior, so it might make sense to keep those together with other schemas.  

I also thought it might be beneficial to see the alternate definitions beside the default to make it more clear that the behavior is not always the same.  

But on reflection I agree that it isn't clear enough.  Options here I see are to add a comment here, put these into the different resource files, or change the separator.  Let me know what you prefer and I'll make a commit to do just that.

File: thousandeyes/schemas.go
Comment (johntdyer) [reply thousandeyes/schemas.go:428]: yea, I agree the double hyphen seems odd

File: thousandeyes/schemas.go
Comment (JTBlanchard) [reply thousandeyes/schemas.go:23]: Based on the above, how would you like me to address this?

Comment (william20111) [reply thousandeyes/schemas.go:23]: Sorry been sleeping on this as I moved companies/roles. Will try to take a look tomorrow. But I would say everything should follow the standard syntax of _ between words. I'm not keen on the double dash.

Comment (JTBlanchard) [reply thousandeyes/schemas.go:23]: No worries, we're comfortable using branches here until then.  I hope your new position is treating you well!

It can definitely be done differently.  It was about delineating between separation between words and separation between identifiers.  But I'm not advocating a particular way of solving it, just noting the difference.  If we end up not distinguishing between the two, nothing breaks.

Comment (william20111) [reply thousandeyes/schemas.go:23]: Hi Sorry for the huge delay on this. If you update to single `-` then im happy to get this merged.

**Comment by johntdyer** [comment]
@JTBlanchard  can you fix the double hyphen so we can merge this ?

**Comment by JTBlanchard** [comment]
As the separator has been changed, is there anything else we need before merging?

**Comment by JTBlanchard** [comment]
Check failure seems to be from artifact signing in the check process rather than a code failure.

```
    ⨯ release failed after 343.03s error=sign failed: {{ .Env.GPG_FINGERPRINT }}: invalid template: template: tmpl:1:7: executing "tmpl" at <.Env.GPG_FINGERPRINT>: map has no entry for key "GPG_FINGERPRINT"
Error: The process '/opt/hostedtoolcache/goreleaser-action/0.159.0/x64/goreleaser' failed with exit code 1
```

**Comment by JTBlanchard** [comment]
@william20111 Once we have this and the other remaining PRs (minus the SDK update) merged, we and some other parties will be ready to put this into production.

With the previously mentioned issue resolved, is there anything else I can or need to do to make these acceptable?

**Comment by william20111** [comment]
lgtm if you bump to latest it will get the CI fix and will merge

**Comment by JTBlanchard** [comment]
@william20111 Thanks!  It all passes now.  I'll stay on top of the other PRs in case any merge conflicts arise as they're applied (though I'm not expecting them as it hasn't arisen in my testing).

**Review by william20111 — APPROVED** [review]

**Pull request merged**
